crates/bevy_ecs/src/world/mod.rs

```diff
@@ -781,7 +791,12 @@ impl World {
     /// # Safety
     /// `component_id` must be valid and correspond to a resource component of type T
     #[inline]
-    unsafe fn insert_resource_with_id<T>(&mut self, component_id: ComponentId, mut value: T) {
+    unsafe fn insert_resource_with_id<T>(
+        &mut self,
+        component_id: ComponentId,
+        mut value: T,
+        overwrite_existing: ComponentCollision,
```

Comment:
Nit: we should use a more generic name than `ComponentCollision`, given that this enum applies to resources as well (and will want to be used for relations down the line).

Reply:
Do you have any suggestions? This issue occured to me but I ultimately settled on ComponentCollision because resources still have to satisfy the <T: Component> constraint (Not that that means much).

Reply:
I think `CollisionBehavior` would be quite clear.

FWIW, I don't expect the component bound to persist there forever; I expect #1843 to be accepted and for us to end up with seperate `Component` and `Resource` traits :)